sklearn/multiclass.py

```diff
@@ -1025,6 +1025,31 @@ def fit(self, X, y):
 
         return self
 
+    def predict_proba(self, X):
+        """Return faux probability estimates.
+
+        Estimates are based upon euclidean distance between the underlying
+        estimators' predictions and the class code.
+
+        Parameters
+        ----------
+        X : (sparse) array-like of shape (n_samples, n_features)
+            Data.
+
+        Returns
+        -------
+        T : (sparse) array-like of shape (n_samples, n_classes)
+            Returns the score of the sample for each class in the model,
+            where classes are ordered as they are in `self.classes_`.
+        """
+        check_is_fitted(self)
+        Y = np.array([_predict_binary(e, X) for e in self.estimators_]).T
+        dist = euclidean_distances(Y, self.code_book_)
+        # Inverse distance weighting:
+        eps = 1e-16
+        proba = (1.0 / (dist + eps)) / np.sum(1.0 / (dist + eps), axis=1)[:, np.newaxis]
```

Comment:
Here, we need to be sure that we handle properly the binary case and the multiclass. For multiclass, we should have an output of `(n_samples, n_classes)` but for the binary case, we will have `(n_samples,)`. The `np.newaxis` makes me think that it would not be the case in the latest case.

It also reveal that we don't have a test for it.

Reply:
Is that so? That's the convention for the `decision_function()` it seems, but not `predict_proba()`. `OneVsRestClassifier()` does not do that, and (at a first glance) no multiclass capable estimators do.

I might have missed something here, some clarification would be greatly appreciated before going further.

Reply:
Oh, right. I confused the shape of `decision_function` and `predict_proba` from the top of the head. So it looks okay then.